test/github_link_test.ml

```diff
@@ -67,13 +80,24 @@ let cases =
       pr_cases "www.github.com" github_repo;
       issue_cases "www.github.com" github_repo;
       commit_cases "www.github.com" github_repo;
+      compare_cases "www.github.com" github_repo;
       pr_cases "example.org" enterprise_repo1;
       issue_cases "example.org" enterprise_repo1;
       commit_cases "example.org" enterprise_repo1;
       pr_cases "example.org/path/to/git" enterprise_repo2;
       issue_cases "example.org/path/to/git" enterprise_repo2;
       commit_cases "example.org/path/to/git" enterprise_repo2;
+      compare_cases "example.org/path/to/git" enterprise_repo2;
       other_cases;
     ]
 
-let () = List.iter cases ~f:(fun (input, expected) -> assert (Poly.equal (gh_link_of_string input) expected))
+let () =
+  List.iter cases ~f:(fun (input, expected) ->
+    assert (
+      match Poly.equal (gh_link_of_string input) expected with
+      | true -> true
+      | false ->
+        Stdio.print_endline input;
```

Comment:
maybe make the message a little bit richer so that it is possible to understand what whent wrong?

Reply:
I removed most of the loggings--sorry I should've cleaned up this commit more

Reply:
but this one is useful? Otherwise when a test fails there are no details and no way to know the link that isn't properly handled

Reply:
gotcha--will add more details!